src/regmod/data.py

```diff
@@ -1,287 +0,0 @@
-        computation. Values of `weights` are required to be between 0 and 1.
-        Default for `col_weights` is `'weights'`. If `col_weights` is not in the
-        data frame, a column with name `col_weights` will be added to the data
-        frame filled with 1.
-    col_offset : str
-        Column name for offset. Same as `weights`, `offset` can be used in
-        computing likelihood. `offset` needs to be pre-transformed according to
-        link function of the parameters. Default for `col_offset` is `'offset'`.
-        If `col_offset` is not in the data frame, a column with name
-        `col_offset` will be added to the data frame filled with 0.
-    df : pd.DataFrame
-        Data frame for the object. Default is an empty data frame.
-    cols : List[str]
-        All the relevant columns, including, `col_obs` (if not `None`),
-        `col_covs`, `col_weights`, `col_offset` and `'trim_weights'`.
-
-    Methods
-    -------
-    is_empty()
-        Indicator of empty data frame.
-    check_cols()
-        Validate if all `self.cols` are in `self.df`.
-    parse_df(df=None)
-        Subset `df` with `self.cols`.
-    fill_df()
-        Automatically add columns `'intercept'`, `col_weights`, `col_offset` and
-        `'trim_weights'`, if they are not present in the `self.df`.
-    detach_df()
-        Set `self.df` to an empty data frame.
-    attach_df(df)
-        Validate `df` and set `self.df=df`.
-    copy(with_df=False)
-        Copy `self` to a new instance of the class.
-    get_cols(cols)
-        Access columns in `self.df`.
-    get_covs(col_covs)
-        Access covariates in `self.df`.
-
-    Notes
-    -----
-    * This class should be replaced by a subclass of a more general dataclass
-    * `get_covs` seems very redundant should only keep `get_cols`.
-    """
-
-    col_obs: Optional[Union[str, List[str]]] = None
-    col_covs: List[str] = field(default_factory=list)
-    col_weights: str = "weights"
-    col_offset: str = "offset"
-    df: DataFrame = field(default_factory=DataFrame)
-    subset_cols: bool = False
-
-    def __post_init__(self):
-        self.col_covs = list(set(self.col_covs).union({'intercept'}))
-        self.cols = self.col_covs + [self.col_weights, self.col_offset, "trim_weights"]
-        if self.col_obs is not None:
-            if isinstance(self.col_obs, str):
-                self.cols.insert(0, self.col_obs)
-            else:
-                self.cols = self.col_obs + self.cols
-
-        if self.is_empty():
-            self.df = DataFrame(columns=self.cols)
-        else:
-            self.parse_df()
-            self.fill_df()
-            self.check_cols()
-
-    def is_empty(self) -> bool:
-        """Indicator of empty data frame.
-
-        Returns
-        -------
-        bool
-            Return `True` when `self.df` is empty.
-        """
-        return self.num_obs == 0
-
-    def check_cols(self) -> None:
```

Comment:
are any of these validations needed, or are they added somewhere else?

Reply:
I think they are definitely needed, will try to add them back.